src/App.tsx

```diff
@@ -1,14 +1,34 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos, selectTodo } from './api';
 
 export const App: React.FC = () => {
+  const [todo, setTodo] = useState<Todo>();
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [appTodo, setAppTodo] = useState<Todo[]>([]);
+  const [check, setCheck] = useState(false);
+  const [selectedTodoId, setSelectedTodoId] = useState<number>(0);
+
+  useEffect(() => {
+    selectTodo(selectedTodoId);
```

Comment:
Ah I understand you re render because your selectedTodo uses the getTodo function again, I see,.